examples/chip-tool-darwin/commands/common/CHIPCommandStorageDelegate.h

```diff
@@ -1,8 +1,13 @@
 #import <CHIP/CHIP.h>
 #import <Foundation/Foundation.h>
 
+NS_ASSUME_NONNULL_BEGIN
+
 @interface CHIPToolPersistentStorageDelegate : NSObject <CHIPPersistentStorageDelegate>
-- (NSString *)CHIPGetKeyValue:(NSString *)key;
-- (void)CHIPSetKeyValue:(NSString *)key value:(NSString *)value;
-- (void)CHIPDeleteKeyValue:(NSString *)key;
+- (nullable NSData *)valueForKey:(NSString *)key;
+// value is nullable because NSKeyValueCoding has a selector collision with us.
+- (BOOL)setValue:(nullable NSData *)value forKey:(NSString *)key;
```

Comment:
Should we say that setting the value to nil is the same as deleting it? (I believe that's the case with `NSKeyValueCoding`.

Reply:
In the protocol definition you can't set the value to nil.

We _could_ change the protocol, but why?  The caller will never pass `nil` when its trying to set...

Or am I missing the point somehow @sagar-apple ?

Reply:
I'm of the opinion to have it look similar to NSDictionary's `setBlah:forKey:` , `blahForKey:` , and `removeBlahForKey:` because the removal is explicit. And since we have a collision with different nullability expectations, at this point it's probably better to use our non-colliding name.

I could change these to `setData:forKey:` , `dataForKey:`, and `removeDataForKey:`, but if we want to avoid more common/generic names, we could change "data" to "storageData" or something similar - so they look like `setStorageData:forKey:`, `storageDataForKey:` and `removeStorageDataForKey:`. That way it's explicitly different and obvious from the method call.

Reply:
Updated naming to avoid the name-collision with NSKeyValueCoding.  Used `setStorageData`/`storageDataForKey`/`removeStorageDataForKey`